src/deluge/io/midi/midi_device_manager.cpp

```diff
@@ -428,29 +428,34 @@ void writeDevicesToFile() {
 	}
 	anyChangesToSave = false;
 
+	bool anyWorthWritting = false;
```

Comment:
this could probably be condensed to a single assignment
```
bool anyWorthWriting = dinMidiPorts.worthWritingToFile() || upstreamUSBMIDICable1.worthWritingToFile() || ...;
if (!anyWorthWriting) {
  for loop{
    anyWorthWriting = true;
    break;
  }
}
if (!anyWorthWriting) {
  f_unlink...
}
```